htdocs/libraries/icms/member/user/Handler.php

```diff
@@ -190,6 +208,13 @@ public function insert(&$user, $force = FALSE) {
 		if ($user->isNew()) {
 			$uid = $this->db->getInsertId();
 			$user->assignVar('uid', $uid);
+			$hash_uid = bin2hex(random_bytes(32));
+			$user->assignVar('hash_uid', $hash_uid);
+			$sql = sprintf(
+				"UPDATE %s SET hash_uid = '%s' WHERE uid = '%u'",
+				$this->db->prefix('users'), $hash_uid, (int) $uid
+			);
+			$result = $this->db->query($sql);
```

Comment:
I'm not sure fully how to solve this but I see possible collision for users with same `hash_uid` problem. 

Maybe here everything should be wrapped into transaction that first tries verify if user with such hash_uid exists and then if not try to update. If fails because of unique index violation, retry again.